app/controllers/tags_controller.rb

```diff
@@ -105,21 +103,32 @@ def show
   end
 
   def feed
-    begin
-      @tag = Tag.find(params[:id])
-    rescue ActiveRecord::RecordNotFound
-      raise ActiveRecord::RecordNotFound, "Couldn't find tag with id '#{params[:id]}'"
-    end
-    @tag = @tag.merger if !@tag.canonical? && @tag.merger
-    # Temp for testing
-    if %w(Fandom Character Relationship).include?(@tag.type.to_s) || @tag.name == 'F/F'
-      if @tag.canonical?
-        @works = @tag.filtered_works.visible_to_all.order('created_at DESC').limit(25)
+    hash = Rails.cache.fetch(Tag.tag_feeds_key(params[:id])) do
```

Comment:
Suggest a comment above this saying that the hash you are building and/or getting from the cache is tag, path, and works, and what each element is used for